src/java/org/apache/cassandra/io/sstable/format/big/BigTableBlockWriter.java

```diff
@@ -0,0 +1,221 @@
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.lifecycle.LifecycleTransaction;
+import org.apache.cassandra.db.rows.UnfilteredRowIterator;
+import org.apache.cassandra.io.FSWriteError;
+import org.apache.cassandra.io.compress.BufferType;
+import org.apache.cassandra.io.sstable.Component;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.SSTable;
+import org.apache.cassandra.io.sstable.SSTableMultiWriter;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.SequentialWriter;
+import org.apache.cassandra.io.util.SequentialWriterOption;
+import org.apache.cassandra.net.async.RebufferingByteBufDataInputPlus;
+import org.apache.cassandra.schema.TableId;
+import org.apache.cassandra.schema.TableMetadataRef;
+
+public class BigTableBlockWriter extends SSTable implements SSTableMultiWriter
+{
+    private final TableMetadataRef metadata;
+    private final LifecycleTransaction txn;
+    private volatile SSTableReader finalReader;
+    private final Map<Component.Type, SequentialWriter> componentWriters;
+
+    private final Logger logger = LoggerFactory.getLogger(BigTableBlockWriter.class);
+
+    private final SequentialWriterOption writerOption = SequentialWriterOption.newBuilder()
+                                                                              .trickleFsync(false)
+                                                                              .bufferSize(2 * 1024 * 1024)
+                                                                              .bufferType(BufferType.OFF_HEAP)
+                                                                              .build();
+    public static final ImmutableSet<Component> supportedComponents = ImmutableSet.of(Component.DATA, Component.PRIMARY_INDEX, Component.STATS,
+                                                                               Component.COMPRESSION_INFO, Component.FILTER, Component.SUMMARY,
+                                                                               Component.DIGEST, Component.CRC);
+
+    public BigTableBlockWriter(Descriptor descriptor,
+                               TableMetadataRef metadata,
+                               LifecycleTransaction txn,
+                               final Set<Component> components)
+    {
+        super(descriptor, ImmutableSet.copyOf(components), metadata,
+              DatabaseDescriptor.getDiskOptimizationStrategy());
+        txn.trackNew(this);
+        this.metadata = metadata;
+        this.txn = txn;
+        this.componentWriters = new HashMap<>(components.size());
+
+        assert supportedComponents.containsAll(components) : String.format("Unsupported streaming component detected %s",
+                                                                           new HashSet(components).removeAll(supportedComponents));
```

Comment:
`Set.removeAll()` returns a `boolean`, so this won't quite work unfortunately.

Reply:
Fixed.